app/src/main/java/org/p2p/wallet/common/EncryptedSharedPreferences.kt

```diff
@@ -0,0 +1,255 @@
+package org.p2p.wallet.common
+
+import androidx.core.content.edit
+import android.content.SharedPreferences
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import timber.log.Timber
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty
+import org.p2p.core.crypto.Hex
+import org.p2p.wallet.common.crypto.keystore.DecodeCipher
+import org.p2p.wallet.common.crypto.keystore.EncodeCipher
+import org.p2p.wallet.common.crypto.keystore.KeyStoreWrapper
+
+class EncryptedSharedPreferences(
+    private val keyStoreWrapper: KeyStoreWrapper,
+    private val sharedPreferences: SharedPreferences,
+    private val gson: Gson
+) {
+    var loggingTag: String = "EncryptedSharedPreferences"
+
+    fun saveString(key: String, data: String) {
+        tryWithLog(key) {
+            val encodedData = keyStoreWrapper.encode(key, data)
```

Comment:
I think, this func should be suspended, and use computation or io dispatcher, cause here we use main thread,
to encode and decode data(and encrypt it)

Reply:
all the Storages and code will be broken if we add some `suspend` stuff here, encoding/decoding is heavy operation, but i didn't see any perfomance issues so far